docs/root/configuration/http/http_filters/credential_injector_filter.rst

```diff
@@ -24,41 +20,60 @@ Configuration
 * This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.credential_injector.v3.CredentialInjector``.
 * :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.credential_injector.v3.CredentialInjector>`
 
-Currently the filter supports :ref:`generic <envoy_v3_api_msg_extensions.http.injected_credentials.generic.v3.Generic>` only.
-Other credential types can be supported as extensions.
+The filter is configured with one of the following supported ``credential_injector`` extensions. Extensions are responsible for fetching the credentials
+from the source. The credentials obtained are then injected into the ``Authorization`` header of the proxied HTTP requests, utilizing either the ``Basic``
+or ``Bearer`` scheme.
+
+Generic Credential Injector
+===========================
+* This extension should be configured with the type URL ``type.googleapis.com/envoy.extensions.http.injected_credentials.generic.v3.Generic``.
+* :ref:`generic <envoy_v3_api_msg_extensions.http.injected_credentials.generic.v3.Generic>`
 
 Here is an example configuration with Generic credential, which injects an HTTP Basic Auth credential into the proxied requests.
 
-.. literalinclude:: _include/credential-injector-filter.yaml
+.. literalinclude:: _include/credential-injector-generic-filter.yaml
     :language: yaml
-    :lines: 28-41
-    :caption: :download:`credential-injector-filter.yaml <_include/credential-injector-filter.yaml>`
+    :lines: 29-39
+    :linenos:
+    :lineno-start: 29
+    :caption: :download:`credential-injector-filter.yaml <_include/credential-injector-generic-filter.yaml>`
 
 
-credential.yaml for Basic Auth:
+Credential which is being used to inject a ``Basic Auth`` credential into the proxied requests:
 
-.. code-block:: yaml
-
-  resources:
-  - "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret"
-    name: credential
-    generic_secret:
-      secret:
-        inline_string: "Basic base64EncodedUsernamePassword"
+.. literalinclude:: _include/credential-injector-generic-filter.yaml
+    :language: yaml
+    :lines: 56-59
+    :linenos:
+    :lineno-start: 56
+    :caption: :download:`credential-injector-filter.yaml <_include/credential-injector-generic-filter.yaml>`
 
-It can also be configured to inject a Bearer token into the proxied requests.
+It can also be configured to inject a ``Bearer`` token into the proxied requests.
 
-credential.yaml for Bearer Token:
+Credential for ``Bearer`` token:
 
 .. code-block:: yaml
 
-  resources:
-  - "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret"
-    name: credential
+  secrets:
+  - name: credential
```

Comment:
rather use a `literalinclude` and limit the lines and/or highlight the important ones

Reply:
realising that this config is not currently in yaml file - but i think there is no reason we cant add it as an alternative secret - ie `credential-bearer` or similiar